pkg/ingester/shipper.go

```diff
@@ -233,41 +230,63 @@ func (s *Shipper) blockMetasFromOldest() (metas []*metadata.Meta, _ error) {
 	return metas, nil
 }
 
-func readShippedBlocks(dir string) (map[ulid.ULID]struct{}, error) {
+func readShippedBlocks(dir string) (map[ulid.ULID]time.Time, error) {
```

Comment:
[nit] We can simplify shipper code a bit if we return `map[ulid.ULID]model.Time` here.